vendor/engines/bulk_email/app/services/bulk_email/bulk_email_searcher.rb

```diff
@@ -7,15 +7,22 @@ class BulkEmailSearcher
     attr_reader :order_details, :search_fields
 
     DEFAULT_SORT = [:last_name, :first_name].freeze
-    SEARCH_TYPES = [:customers, :account_owners, :customers_and_account_owners, :authorized_users].freeze
+    USER_TYPES = %i(customers authorized_users account_owners).freeze
 
     def initialize(search_fields)
       @search_fields = search_fields
     end
 
+    def user_types
+      @user_types ||=
+        USER_TYPES & (search_fields[:bulk_email].try(:[], :user_types) || []).map(&:to_sym)
```

Comment:
I see this as somewhat related to the blank `@user` issue; the try exists because there's otherwise no way to tell the difference between hitting the search form landing page and actually doing a search. I have a patch coming up.